grav/message.go

```diff
@@ -45,6 +48,13 @@ type Message interface {
 	Unmarshal([]byte) error
 }
 
+type messageWithOrigin struct {
+	Message
+
+	// originUUID is the UUID of the pod from which the message originated.
+	originUUID string
+}
+
```

Comment:
Introduced this to not change the `Message` interface, which would be a breaking change because the interface is public.